XenModel/Actions/AsyncAction.cs

```diff
@@ -179,7 +179,7 @@ public void RunAsync(SudoElevationResult sudoElevationResult)
         /// use that session for the action: it is then the responsibility of the calling function
         /// to make sure the session has the appropriate privileges and tidy it up afterwards.
         /// </summary>
-        public void RunExternal(Session session)
+        public void RunSync(Session session)

```

Comment:
nit: we could set `session = null` as a default. It's used in a few places.

Even nit-tier: The comment says "run this action non-async". We could just write "synchronously"

Reply:
Will fix the comment. However, I don't see RunSync being called anywhere with a null session.

Reply:
To be fair, it's only used three times. That's why it's "nit":
![image](https://user-images.githubusercontent.com/32554698/162916992-703c805a-3632-4e43-bf78-4801bee66651.png)